presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveCoercion.java

```diff
@@ -140,6 +146,24 @@
                         "STORED AS AVRO");
     }
 
+    enum ExecutorType
+    {
+        PRESTO(onPresto()),
+        HIVE(onHive());
```

Comment:
`onPresto()` could return something that's scoped for a test method or class.
(maybe it does already?)

thus do it lazy, only when `getExecutor` is called.
or, do like here
https://github.com/prestosql/presto/blob/34504af3ce2a4dfff4fecdb1d26d840787345220/presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java#L793-L797
where the enum is dumb